### ISSUE TITLE
types: Add generic types to Container-like views

### ISSUE THREAD

**Issue by CatchABus** [pixijs/ui#175]
**Please Describe The Problem To Be Solved**
It would be a nice addition if DisplayObjects were more type-friendly when it comes to handling children.
Pixi itself does that with `Container`.

```ts
const container = new Container<Sprite>();
// Now I can manipulate children without casting type
```

**(Optional): Suggest A Solution**
Example using a `List`
```ts
const list = new List<Graphics>({
    children: [
        new Graphics().beginFill(0x000000).drawRect(0, 0, 50, 50),
        new Graphics().beginFill(0xFFFFFF).drawRect(0, 0, 50, 50),
    ],
 });

console.log(list.children); // Type of Graphics[]
```

**Comment by CyberDex** [comment]
You are welcome to create this PR ;)

**Comment by CatchABus** [comment]
Closed via #191
